plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityAnger.java

```diff
@@ -3,115 +3,81 @@
 import com.denizenscript.denizen.objects.EntityTag;
 import com.denizenscript.denizencore.objects.core.DurationTag;
 import com.denizenscript.denizencore.objects.Mechanism;
-import com.denizenscript.denizencore.objects.ObjectTag;
-import com.denizenscript.denizencore.objects.properties.Property;
-import com.denizenscript.denizencore.objects.properties.PropertyParser;
 import org.bukkit.entity.Bee;
-import org.bukkit.entity.Entity;
 import org.bukkit.entity.PigZombie;
 
-public class EntityAnger implements Property {
+public class EntityAnger extends EntityProperty<DurationTag> {
 
-    public static boolean describes(ObjectTag entity) {
-        if (!(entity instanceof EntityTag)) {
-            return false;
-        }
-        Entity bukkitEntity = ((EntityTag) entity).getBukkitEntity();
-        return bukkitEntity instanceof PigZombie
-                || bukkitEntity instanceof Bee;
-    }
+    // <--[property]
+    // @object EntityTag
+    // @name anger
+    // @input DurationTag
+    // @description
+    // Controls the remaining anger time of a PigZombie or Bee.
+    // -->
 
-    public static EntityAnger getFrom(ObjectTag entity) {
-        if (!describes(entity)) {
-            return null;
-        }
-        else {
-            return new EntityAnger((EntityTag) entity);
-        }
+    public static boolean describes(EntityTag entity) {
+        return entity.getBukkitEntity() instanceof Bee
+                || entity.getBukkitEntity() instanceof PigZombie;
     }
 
-    public static final String[] handledMechs = new String[] {
-            "anger"
-    };
+    EntityTag entity;
 
-    public EntityAnger(EntityTag entity) {
-        this.entity = entity;
+    @Override
+    public DurationTag getPropertyValue() {
+        return new DurationTag((long) getAnger());
     }
 
-    EntityTag entity;
-
     @Override
-    public String getPropertyString() {
-        return new DurationTag((long) getAnger()).identify();
+    public void setPropertyValue(DurationTag param, Mechanism mechanism) {
+        if (mechanism.requireObject(DurationTag.class)) {

```

Comment:
Same here, no need for `requireObject`/`valueAsType`, since `autoRegister` does all of that internally & you already get a valid and checked `DurationTag` here